### PR TITLE
Add sample for hosting a component using nethost and hostfxr

### PR DESCRIPTION
**Pull request by elinor-fung**
## Summary

Add component hosting sample using `nethost` and `hostfxr` interfaces

cc @sdmaclea @vitek-karas @lpereira @AaronRobinsonMSFT

### CODE REVIEW

**Comment by AaronRobinsonMSFT** [comment]
cc @richlander

File: core/hosting/HostWithHostFxr/readme.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+Demonstrates how to locate and initialize .NET Core 3.0 from a non-.NET Core process and subsequently load and call into  a .NET Core assembly.
+
+The `nethost` header and library are part of the Microsoft.NETCore.DotNetAppHost package and are also installed as a runtime pack by the .NET SDK. The library should be deployed alongside the host. This sample uses the files installed with the .NET SDK. 
+
+The `coreclr_delegates.h` and `hostfxr.h` files are copied from the [core-setup](https://github.com/dotnet/core-setup) repo. 
+
+Addtional comments are contained in source and project files.
+
+Requirements
```
Comment (rpetrusha) [inline core/hosting/HostWithHostFxr/readme.md]: perhaps name this "Prerequisites", to differentiate prerequisites from the two items that are also required?

File: core/hosting/HostWithHostFxr/readme.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+* .NET Core 3.0 (at least Preview 6) - [https://dot.net](https://github.com/dotnet/core-sdk#installers-and-binaries)
+
+* C++ compiler
+  * Windows: `cl.exe`
+  * Linux/OSX: `g++`
+
+Build and Run
+-------------
+
+1) In order to build and run, all requirements must be installed. The is also required:
```
Comment (rpetrusha) [inline core/hosting/HostWithHostFxr/readme.md]: "all requirements" becomes "all prerequisites" if you make the previous change.

```suggestion
1) In order to build and run, all requirements must be installed. The following are also required:
```

**Comment by rpetrusha** [comment]
This looks good, @elinor-fung. Thanks for making the additional changes.  I'll merge your PR now.

**Pull request merged**